src/main/java/nextstep/session/domain/Session.java

```diff
@@ -5,64 +5,105 @@
         this.sessionStatus = DEFAULT_SESSION_STATUS;
+        this.sessionRecruitStatus = DEFAULT_RECRUIT_STATUS;
         this.sessionType = sessionType;
     }
 
-    public Session(Long id, LocalDateTime createdAt, LocalDateTime updatedAt, Long creatorId, LocalDate startDate, LocalDate endDate, SessionImage sessionImage, SessionStatus sessionStatus, SessionType sessionType) {
-        super(id, createdAt, updatedAt);
+    public Session(Long id, LocalDateTime createdAt, LocalDateTime updatedAt, Long creatorId, LocalDate startDate, LocalDate endDate, SessionImage sessionImage, SessionStatus sessionStatus, SessionRecruitStatus sessionRecruitStatus, SessionType sessionType, List<Enrollment> enrollments, List<Admission> admissions) {
+        super(createdAt, updatedAt);
+        this.id = id;
         this.creatorId = creatorId;
         this.sessionDate = new SessionDate(startDate, endDate);
         this.sessionImage = sessionImage;
         this.sessionStatus = sessionStatus;
+        this.sessionRecruitStatus = sessionRecruitStatus;
         this.sessionType = sessionType;
+        this.enrollments = new Enrollments(enrollments);
+        this.admissions = new Admissions(admissions);
     }
 
     @Override
-    public void enroll(NsUser user) {
+    public Enrollment enroll(NsUser user) {
         validateStatus();
         validateCommonEnroll(user);
-        enrollments.add(user);
+        return enrollments.add(user, this);
     }
 
     private void validateStatus() {
-        if (sessionStatus != SessionStatus.RECRUITING) {
+        if (sessionStatus == SessionStatus.END) {
+            throw new IllegalStateException("종료된 강의는 신청 불가능합니다.");
+        }
+        if (sessionRecruitStatus != SessionRecruitStatus.OPEN) {
             throw new IllegalStateException("모집중인 강의만 신청 가능합니다.");
         }
     }
 
-    protected void validateCommonEnroll(NsUser nsUser) {
-    }
+    abstract protected void validateCommonEnroll(NsUser nsUser);
 
     @Override
     public int enrolledNumber() {
         return enrollments.enrolledNumber();
     }
 
+    public Enrollment admiss(NsUser loginUser, NsUser student) {
+        if (!creatorId.equals(loginUser.getId())) {
+            throw new IllegalArgumentException("강사만 승인할 수 있습니다.");
+        }
+        if (admissions.isAdmiss(student, this)) {
+            throw new IllegalArgumentException("선발되지 않은 학생입니다.");
+        }
+        return enrollments.admiss(student, this);
+    }
+
+    public Enrollment cancel(NsUser loginUser, NsUser student) {
+        if (!creatorId.equals(loginUser.getId())) {
+            throw new IllegalArgumentException("강사만 승인할 수 있습니다.");
+        }
+        return enrollments.cancel(student, this);
+    }
+
     @Override
     public void changeStatus(SessionStatus status) {
         sessionStatus = status;
     }
 
+    public void changeRecruit(SessionRecruitStatus recruitStatus) {
+        if (recruitStatus.equals(SessionRecruitStatus.OPEN)) {
```

Comment:
```suggestion
        if (recruitStatus.isOpen()) {
```
위와 같이 enum도 클래스와 같이 메시지를 보내 구현할 수 있다.

Reply:
반영했습니다!